go/worker/common/p2p/p2p.go

```diff
@@ -362,6 +365,73 @@ func New(identity *identity.Identity, consensus consensus.Backend) (*P2P, error)
 		return nil, fmt.Errorf("worker/common/p2p: failed to create connection gater: %w", err)
 	}
 
+	// Block peers specified in the blacklist.
+	blacklist := viper.GetStringSlice(CfgP2PBlockedPeerIPs)
+	for _, blockedIP := range blacklist {
+		parsedIP := net.ParseIP(blockedIP)
+		if parsedIP == nil {
+			return nil, fmt.Errorf("worker/common/p2p: malformed blocked IP: %s", blockedIP)
+		}
+
+		if grr := cg.BlockAddr(parsedIP); grr != nil {
+			return nil, fmt.Errorf("worker/common/p2p: failed to block IP (%s): %w", blockedIP, err)
+		}
+	}
+
+	// Maintain persistent peers.
+	persistentPeers := make(map[core.PeerID]bool)
+	persistentPeersAI := []peer.AddrInfo{}
+	for _, pp := range viper.GetStringSlice(CfgP2PPersistentPeers) {
+		// The persistent peer addresses are in the format P2Ppubkey@IP:port,
+		// because we use a similar format elsewhere and it's easier for users
+		// to understand than a multiaddr.
+
+		if strings.Count(pp, "@") != 1 || strings.Count(pp, ":") != 1 {
+			return nil, fmt.Errorf("worker/common/p2p: malformed persistent peer address (expected P2Ppubkey@IP:port)")
+		}
+
+		pkaddr := strings.Split(pp, "@")
+		pubkey := pkaddr[0]
+		addr := pkaddr[1]
+
+		var pk signature.PublicKey
+		if grr := pk.UnmarshalText([]byte(pubkey)); grr != nil {
+			return nil, fmt.Errorf("worker/common/p2p: malformed persistent peer address: cannot unmarshal P2P public key (%s): %w", pubkey, grr)
+		}
+		pid, grr := PublicKeyToPeerID(pk)
+		if grr != nil {
+			return nil, fmt.Errorf("worker/common/p2p: invalid persistent peer public key (%s): %w", pubkey, grr)
+		}
+
+		ipport := strings.Split(addr, ":")
```

Comment:
Should probably use [`net.SplitHostPort`](https://pkg.go.dev/net#SplitHostPort).

Reply:
Nice, thanks!